bug: TestRunSessionCreateMissingRequiredFields fails on main

## Description

`TestRunSessionCreateMissingRequiredFields` in `skills/pi-rpc/scripts/cmd/pi-cli` fails on `main`. The test expects an error when `provider` or `model` is not set, but the validation is not implemented (or has regressed).

```
--- FAIL: TestRunSessionCreateMissingRequiredFields (0.20s)
    session_test.go:83: expected error when provider/model not set
FAIL    github.com/nq-rdl/agent-skills/skills/pi-rpc/scripts/cmd/pi-cli    0.234s
```

## Discovery

Surfaced when attempting to push a Makefile-only fix (PR #69). The pre-push hook runs `go test -race ./...`, which caught the failure. The fix in PR #69 was pushed with `--no-verify` because the test failure pre-dates those changes (verified by stashing and running the test against unmodified `main`).

## Impact

- The pre-push hook is silently bypassed by anyone who sees this failure and uses `--no-verify`
- Required-field validation for `pi-cli session create` may not be enforced at runtime

## Steps to reproduce

```bash
cd skills/pi-rpc/scripts
go test ./cmd/pi-cli/... -run TestRunSessionCreateMissingRequiredFields -v
```

## Expected

Test passes — `pi-cli session create` returns an error when `--provider` or `--model` is omitted.

## Suggested fix

Check `session_test.go:83` — either the validation was removed from the `create` command handler and needs to be restored, or the test expectation no longer matches the current CLI flag structure.

--- a/.changes/unreleased/Fixed-20260419-061142-pi-cli-session-create-defaults-test.yaml
+++ b/.changes/unreleased/Fixed-20260419-061142-pi-cli-session-create-defaults-test.yaml
@@ -1,0 +1,3 @@
+kind: Fixed
+body: pi-cli `session create` test coverage — add positive-case `TestRunSessionCreateDefaultsAccepted` asserting empty provider/model are forwarded verbatim so the server can apply PI_DEFAULT_PROVIDER / PI_DEFAULT_MODEL (closes #70)
+time: 2026-04-19T06:11:42Z

--- a/skills/pi-rpc/scripts/cmd/pi-cli/session_test.go
+++ b/skills/pi-rpc/scripts/cmd/pi-cli/session_test.go
@@ -84,6 +84,37 @@ func TestRunSessionCreateUnreachableServer(t *testing.T) {
 	}
 }
 
+func TestRunSessionCreateDefaultsAccepted(t *testing.T) {
+	// Issue #70: empty provider/model must be forwarded verbatim so the server
+	// can apply PI_DEFAULT_PROVIDER / PI_DEFAULT_MODEL. The CLI must not
+	// reject or mutate empty values.
+	var receivedProvider, receivedModel string
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req map[string]any
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			t.Errorf("decode request body: %v", err)
+		}
+		receivedProvider, _ = req["provider"].(string)
+		receivedModel, _ = req["model"].(string)
+		w.Header().Set("Content-Type", "application/json")
+		_ = json.NewEncoder(w).Encode(map[string]any{
+			"sessionId": "defaults-abc",
+			"state":     "SESSION_STATE_IDLE",
+		})
+	}))
+	defer srv.Close()
+
+	if err := runSessionCreate(context.Background(), srv.URL, "", "", "/tmp", "", 0); err != nil {
+		t.Fatalf("runSessionCreate with empty provider/model failed: %v", err)
+	}
+	if receivedProvider != "" {
+		t.Errorf("provider forwarded to server = %q, want empty string", receivedProvider)
+	}
+	if receivedModel != "" {
+		t.Errorf("model forwarded to server = %q, want empty string", receivedModel)
+	}
+}
+
 func TestRunSessionList(t *testing.T) {
 	srv := newTestServer(t, map[string]any{
 		"/pirpc.v1.SessionService/List": map[string]any{

--- a/skills/pi-rpc/scripts/cmd/pi-cli/session_test.go
+++ b/skills/pi-rpc/scripts/cmd/pi-cli/session_test.go
@@ -89,10 +89,13 @@ func TestRunSessionCreateDefaultsAccepted(t *testing.T) {
 	// can apply PI_DEFAULT_PROVIDER / PI_DEFAULT_MODEL. The CLI must not
 	// reject or mutate empty values.
 	var receivedProvider, receivedModel string
+	done := make(chan struct{})
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer close(done)
 		var req map[string]any
 		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
 			t.Errorf("decode request body: %v", err)
+			return
 		}
 		receivedProvider, _ = req["provider"].(string)
 		receivedModel, _ = req["model"].(string)
@@ -104,9 +107,10 @@ func TestRunSessionCreateDefaultsAccepted(t *testing.T) {
 	}))
 	defer srv.Close()
 
-	if err := runSessionCreate(context.Background(), srv.URL, "", "", "/tmp", "", 0); err != nil {
+	if err := runSessionCreate(context.Background(), srv.URL, "", "", t.TempDir(), "", 0); err != nil {
 		t.Fatalf("runSessionCreate with empty provider/model failed: %v", err)
 	}
+	<-done
 	if receivedProvider != "" {
 		t.Errorf("provider forwarded to server = %q, want empty string", receivedProvider)
 	}

